File: cpp/Maximize_Area_of_Square_Hole_in_Grid_2943.cpp
class Solution {
public:
    int maximizeSquareHoleArea(int n, int m, vector<int>& hBars, vector<int>& vBars) {
        sort(hBars.begin(), hBars.end());
        int prevHInd = 0, maxHCont = 0;
        for (int i = 0; i < hBars.size(); i++) {
            if (hBars[i] - hBars[prevHInd] == i - prevHInd) {
                maxHCont = max(maxHCont, i - prevHInd + 1);
                continue;
            }
            else {
                prevHInd = i;
            }
        }
        sort(vBars.begin(), vBars.end());
        int prevVInd= 0, maxVCont = 0;
        for (int i = 0; i < vBars.size(); i++) {
            if (vBars[i] - vBars[prevVInd] == i - prevVInd) {
                maxVCont = max(maxVCont, i - prevVInd + 1);
                continue;
            }
            else {
                prevVInd = i;
            }
        }
        int side = min(maxHCont, maxVCont) + 1;
        return side * side;
    }
};

File: python/Maximize_Area_of_Square_Hole_in_Grid_2943.py
class Solution:
    def maximizeSquareHoleArea(self, n: int, m: int, hBars: List[int], vBars: List[int]) -> int:
        prevHInd, maxHCount = 0, 0
        hBars.sort()
        for ind, bar in enumerate(hBars):
            if hBars[ind] - hBars[prevHInd] == ind - prevHInd:
                maxHCount = max(maxHCount, ind - prevHInd + 1)
            else:
                prevHInd = ind
        prevVInd, maxVCount = 0, 0
        vBars.sort()
        for ind, bar in enumerate(vBars):
            if vBars[ind] - vBars[prevVInd] == ind - prevVInd:
                maxVCount = max(maxVCount, ind - prevVInd + 1)
            else:
                prevVInd = ind
        side = min(maxHCount, maxVCount) + 1
        return side ** 2
